bitswap/src/behaviour.rs

```diff
@@ -289,9 +311,16 @@ impl NetworkBehaviour for Bitswap {
     }
 
     #[allow(clippy::type_complexity)]
-    fn poll(&mut self, ctx: &mut Context, _: &mut impl PollParameters)
-        -> Poll<NetworkBehaviourAction<<<Self::ProtocolsHandler as IntoProtocolsHandler>::Handler as ProtocolsHandler>::InEvent, Self::OutEvent>>
-    {
+    fn poll(
+        &mut self,
+        ctx: &mut Context,
+        _: &mut impl PollParameters,
+    ) -> Poll<
+        NetworkBehaviourAction<
+            Self::OutEvent,
+            <Self::ConnectionHandler as IntoConnectionHandler>::Handler,
+        >,
```

Comment:
No, it didn't resolve the issue. I added that commit (and hope the types were equivalent because it would confuse me even more). Just the `ipfs` crate has a build issue.

Reply:
I would run a cargo expand on the crate but I know far too little about this crate and libp2p to be helpful.